### ISSUE TITLE
Add "wheelchair accessible" checkbox to hosting offers

### ISSUE THREAD

**Issue by simison** [Trustroots/trustroots#736]
- Add "wheelchair accessible" checkbox to hosting offers
- Add possibility to filter search results by it 

Feel free to list in comments below if there could be some other filters like this to help out folks who need good accessibility.

Some more ideas might be found by studying Accomable: https://en.wikipedia.org/wiki/Accomable

**Comment by petr-hajek** [comment]
In which section of offer should the checkbox be? Description or Availability? I'd also add similar filter for pet-friendly offers - a travelling with dogs,etc. can be tricky.

**Comment by simison** [comment]
Pet-friendliness seems like a tribe, just like "Family" tribe is?

Accessibility, on the other hand, is a limit of physical space and not dependant on the person who is hosting.

**Comment by simison** [comment]
>In which section of offer should the checkbox be? Description or Availability?

Availability seems good:

<img width="389" alt="image" src="https://user-images.githubusercontent.com/87168/47380713-bb7b5d80-d706-11e8-8b64-a3e1cf1ea411.png">

**Comment by Regenwasser** [comment]
```
<label tr-switch> 
<input type="checkbox" 
```

Toggles/switches are used like a light switch - only if it's an instantaneous action. Here in the profile settings we still have to hit "Save & Exit", so a checkbox is commonly used.

> Nevertheless, this should be separate issue.

I think this is important. The question is if we will make a single issue for every option we want to add? Pets, Wheelchair Accessibility, Smoking, Sleeping Arrangements. Why not all in one go?

**Comment by simison** [comment]
>Toggles/switches are used like a light switch - only if it's an instantaneous action. Here in the profile settings we still have to hit "Save & Exit", so a checkbox is commonly used.

Hmm yeah, fair point. My motivation to use it here was because we don't have visually nice checkbox, we have just:

<img width="262" alt="image" src="https://user-images.githubusercontent.com/87168/47389536-b37ae800-d71d-11e8-9d2f-1ca576dc54ff.png">

Let's go with checkbox instead for consistency in experience. Here's the html structure we're trying to follow: https://getbootstrap.com/docs/3.3/css/#checkboxes-and-radios

**Comment by simison** [comment]
Let's keep the interface more simple for now and add only "wheelchair accessible". 

Adding new filters like "smokers" might be valuable to few but the penalty from the clutter is big for many.

I don't expect lots of use from "wheelchair accessible" neither but it has value for different reasons.

**Comment by guaka** [comment]
"Smoking inside" is definitely an accessibility issue, should have a
checkbox for that. And yes, most people won't come back to these settings
once they have a profile. So it's good to have this stuff asap.

**Comment by simison** [comment]
I'm still a bit on the edge on this but leaning towards adding some of these after the discussion!

@Regenwasser based on the info you shared it sounds like we _shouldn't_ add the wheelchair accessible filter. :-D ...or at least put quite some thought on wording?

> Smoking (like Pets, Sleeping Arrangements) can lead to bad experiences for guests and hosts. If smoking is allowed indoors then it's usually a chain-smoker that lives in that flat. Not just a few that would rather stay in a hostel than with such a host.

Wheelchair accessibility seems like the only filter that people would use to _include_ only wheelchair accessible places, while the rest seem like one would want to use them to _exclude_ search results. These filters would require users to _always_ check them, which again leads to bad UX & clutter.

What do you think about changing the UI so that folks would be nudged to write info about pets and sleeping arrangements and such to their hosting description instead of using it as a filter? Something like this:

<img width="1204" alt="image" src="https://user-images.githubusercontent.com/87168/47419116-2cad2600-d784-11e8-90f4-6a1ca87e3ffa.png">

That way we wouldn't be forced to reduce these things into yes/no values, but would let people elaborate more.
Especially "sleeping arrangements" seems like something that would be hard to pin down to searchable values, considering how much they vary from culture to culture.

That said, wheelchair accessibility still seems like a good candidate for a filter instead.

**Comment by petr-hajek** [comment]
Considering what @Regenwasser said, if there is such checkbox it should have a toolbox over _wheelchair accessible_  explaining what the user should think about when evaluating his place. Also there should be explicitly pointed out to elaborate more on the accessibility in the sleeping conditions section.

**Comment by Regenwasser** [comment]
> @Regenwasser based on the info you shared it sounds like we shouldn't add the wheelchair accessible filter. :-D ...or at least put quite some thought on wording?

Yes, I looked a bit into it and it seems it won't be very useful like that. Maybe with different wording/implemetation. I do not think that something like a tooltip would be enough as not everyone reads them. If we want to include something for wheelchair users we should reach out to that FB group. 

> What do you think about changing the UI so that folks would be nudged to write info about pets and sleeping arrangements 

Here I see the same issue that you had with text references (and why you came up with reference tags). People don't use it properly and just fill in the minimum amount. If it's in the text, every profile has to be opened one by one. With values it can be searchable - and therefore be useful. 

I don't see why that is related to bad UX. Using search filters is optional and in the hosting settings that most users just set up once something like a checkbox is faster and easier, so more people will do it - especially those that keep their profiles quite empty. 

Trustroots isn't just filled with ultra-relaxed people who are fine with everything - there are all kinds of different users. Bad UX would be the opposite. Imagine the painful experience of an e-commerce or hotel site had the important info in descriptions and filters wouldn't exist. 

On eBay, a tinkerer (similiar to Trustroot founders/power users) who's hunting for spare parts might be fine with _almost all kinds of conditions_ of technical products (like Trustroots founders might be fine with all kinds of sleeping arrangements), while others would be very frustrated if they'd have to open every article to read the descriptions, send messages or look at all pictures to find out. 

> Especially "sleeping arrangements" seems like something that would be hard to pin down to searchable values, considering how much they vary from culture to culture. That way we wouldn't be forced to reduce these things into yes/no values, but would let people elaborate more.

Can expand a bit on the cultural issues here? I think the way it's usually done is good: 

- Shared sleeping surface
- Shared room
- Public room 
- Private room

Most users are not in or from areas where it varies much and our site is in English only. We have most users in Europe/Germany and in big cities, too.

**Comment by guaka** [comment]
(I don't really want to stay with (or even visit) people who smoke inside their home, so ideally I'd have that "smoking inside" filter set to "no" all the time.)

**Comment by topalavlad** [comment]
> (I don't really want to stay with (or even visit) people who smoke inside their home, so ideally I'd have that "smoking inside" filter set to "no" all the time.)

Couldn't preferences like these be saved in the local storage? This way the user would have to check it only once per browser.
